todo-list/src/api/auth.ts

```diff
@@ -0,0 +1,50 @@
+import { LoginResponse, RegisterResponse } from "../types/auth/auth.type";
+import { safeFetch } from "./safeFetch";
+
+export async function login(
+  username: string,
+  password: string,
+): Promise<LoginResponse> {
+  const url = `${import.meta.env.VITE_API_URL}/user/login`;

```

Comment:
host should not be passed into safeFetch every time but implemented inside the function itself